og_ui/src/Form/OgPermissionsForm.php

```diff
@@ -0,0 +1,338 @@
+
+    foreach ($permissions_by_provider as $provider => $permissions) {
+      // Skip empty permission provider groups.
+      if (empty($permissions)) {
+        continue;
+      }
+
+      // Provider.
+      $form['permissions'][$provider] = [
+        [
+          '#wrapper_attributes' => [
+            'colspan' => count($roles) + 1,
+            'class' => ['module'],
+            'id' => 'provider-' . Html::getId($provider),
+          ],
+          '#markup' => $provider,
+        ],
+      ];
+
+      /** @var \Drupal\og\Permission $permission */
+      foreach ($permissions as $perm => $permission) {
+        // Fill in default values for the permission.
+        $perm_item = [
+          'title' => $permission->getTitle(),
+          'description' => $permission->getDescription(),
+          'restrict access' => $permission->getRestrictAccess(),
+          'warning' => $permission->getRestrictAccess() ? $this->t('Warning: Give to trusted roles only; this permission has security implications.') : '',
+        ];
+
+        $form['permissions'][$perm]['description'] = [
+          '#type' => 'inline_template',
+          '#template' => '<div class="permission"><span class="title">{{ title }}</span>{% if description or warning %}<div class="description">{% if warning %}<em class="permission-warning">{{ warning }}</em> {% endif %}{{ description }}</div>{% endif %}</div>',
+          '#context' => [
+            'title' => $perm_item['title'],
+          ],
+        ];
+
+        // Show the permission description.
+        if (!$hide_descriptions) {
+          $form['permissions'][$perm]['description']['#context']['description'] = $perm_item['description'];
+          $form['permissions'][$perm]['description']['#context']['warning'] = $perm_item['warning'];
+        }
+
+        foreach ($roles as $rid => $role) {
+          list(,, $rid_simple) = explode('-', $rid, 3);
+
+          // The roles property indicates which roles the permission applies to.
+          $permission_applies = TRUE;
+          if (property_exists($permission, 'roles')) {
+            $target_roles = $permission->get('roles');
+            $permission_applies = empty($target_roles) || in_array($rid_simple, $target_roles);
+          }
+
+          if ($permission_applies) {
+            $form['permissions'][$perm][$rid] = [
+              '#title' => $role->getName() . ': ' . $perm_item['title'],
+              '#title_display' => 'invisible',
+              '#wrapper_attributes' => [
+                'class' => ['checkbox'],
+              ],
+              '#type' => 'checkbox',
+              '#default_value' => $role->hasPermission($perm) ? 1 : 0,
+              '#attributes' => ['class' => ['rid-' . $rid, 'js-rid-' . $rid]],
+              '#parents' => [$rid, $perm],
+            ];
+
+            // Show a column of disabled but checked checkboxes.
+            // Only applies to admins or default roles.
+            if ($roles[$rid]->get('is_admin') ||
```

Comment:
Here it is. I found it. I don't think we can safely assume that someone won't want to override this. Can we remove this fancy logic?

Reply:
Diving into the code, this isn't possible. We have an is_admin thing on roles. I'll just create a new 'Group Manager' role that has the access I want.